numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1794,10 +1816,8 @@ unpack_bits(PyObject *input, int axis, PyObject *count_obj)
             v |= (npy_uint64)((j &  32) ==  32) << 40;
             v |= (npy_uint64)((j &  64) ==  64) << 48;
             v |= (npy_uint64)((j & 128) == 128) << 56;
-#if NPY_BYTE_ORDER == NPY_LITTLE_ENDIAN
-            v = npy_bswap8(v);
-#endif
-            unpack_lookup_64[j] = v;
+            unpack_lookup_64l[j] = v;
+            unpack_lookup_64b[j] = npy_bswap8(v);
```

Comment:
Why are we storing this in a `uint64_t` with indeterminate endianness in the first place?

Why not just compute this byte-by-byte? Micro-optimizing a one-off operation over 256 item seems awfully premature here, and makes the if below harder to follow.

Also, is there any guarantee that `unpack_lookup_64l` will have the correct alignment requirements to make this legal?

I realize that all of this criticism applies equally to the code that is already here

Reply:
ok, removed all the code that deals with lookup